pkg/gofr/shutdown.go

```diff
@@ -0,0 +1,26 @@
+package gofr
+
+import (
+	"context"
+	"errors"
+)
+
+// shutdownWithTimeout handles the shutdown process with context timeout.
+func shutdownWithTimeout(ctx context.Context, shutdownFunc func(ctx context.Context) error, forceCloseFunc func() error) error {
```

Comment:
```suggestion
// shutdownWithContext handles the shutdown process with context timeout.
func shutdownWithContext(ctx context.Context, shutdownFunc func(ctx context.Context) error, forceCloseFunc func() error) error {
```

There is no timeout in the method

Or maybe


```suggestion
// shutdownRespectContextExpiration handles the shutdown process with context timeout.
func shutdownRespectContextExpiration(ctx context.Context, shutdownFunc func(ctx context.Context) error, forceCloseFunc func() error) error {
```

Wording is a pain

Reply:
I'm fine with it